tests/pamiq_core/threads/test_thread_control.py

```diff
@@ -154,27 +154,50 @@ def handler(self, read_only_controller) -> ControllerCommandHandler:
         return ControllerCommandHandler(read_only_controller)
 
     def test_stop_if_pause_when_already_resumed(
-        self, thread_controller: ThreadController, handler: ControllerCommandHandler
+        self,
+        thread_controller: ThreadController,
+        handler: ControllerCommandHandler,
+        mocker,
     ) -> None:
+        # prepare mock objects
+        on_paused_callback_mock = mocker.patch.object(handler, "_on_paused")
+        on_resumed_callback_mock = mocker.patch.object(handler, "_on_resumed")
+
         # immediately return if already resumed
         thread_controller.resume()
         start = time.perf_counter()
         handler.stop_if_pause()
         assert time.perf_counter() - start < 1e-3
 
+        # callbacks are not called
+        on_paused_callback_mock.assert_not_called()
+        on_resumed_callback_mock.assert_not_called()
+
     def test_stop_if_pause_pause_to_resume(
-        self, thread_controller: ThreadController, handler: ControllerCommandHandler
+        self,
+        thread_controller: ThreadController,
+        handler: ControllerCommandHandler,
+        mocker,
     ) -> None:
+        # prepare mock objects
+        on_paused_callback_mock = mocker.patch.object(handler, "_on_paused")
+        on_resumed_callback_mock = mocker.patch.object(handler, "_on_resumed")
+
         # immediately return if resumed after waiting
         thread_controller.pause()
         threading.Timer(0.1, thread_controller.resume).start()
         start = time.perf_counter()
         handler.stop_if_pause()
         assert 0.1 <= time.perf_counter() - start < 0.2
 
+        # callbacks are called
+        on_paused_callback_mock.assert_called_once()
+        on_resumed_callback_mock.assert_called_once()
```

Comment:
`_with`までつけると引数を与えないことまで検証できます。
```suggestion
        on_paused_callback_mock.assert_called_once_with()
        on_resumed_callback_mock.assert_called_once_with()
```

Reply:
おっしゃるとおりです！
（前回も同じことを言われた気がします、、、すみません！　以降は基本的に with ありを利用します！）